src/compo/viirs_l1bnc2ioda.py

```diff
@@ -158,6 +166,9 @@ def _read(self):
                 orbit_ad[:] = 1
             self.varAttrs['satelliteAscendingFlag', metaDataName]['description'] = '0=descending, 1=ascending'
 
+            # secant of solar zenith angle to get true reflectance (apply_secterm option)
+            sec_term = np.where(solar_za != 90, 1. / np.cos(solar_za * np.pi / 180.), 1.)
```

Comment:
and done,, this is the way to handle the case for an array `solar_za`

Reply:
@weiwilliam please let me know if this doesn't work for you

I'm getting it to run with the `secterm` applied and matching the output file of the branch

Reply:
Thank you @BenjaminRuston, I'm also working on it.

Reply:
@BenjaminRuston It works well and the output passed the nccmp.